numba/pycc/cc.py

```diff
@@ -1,6 +1,9 @@
-from distutils import dir_util, log
-from distutils.command import build_ext
-from distutils.extension import Extension
+from setuptools import distutils as dutils
+dir_util = dutils.dir_util
+log = dutils.log
```

Comment:
```suggestion
from setuptools.distutils import dir_util, log
```

Reply:
Thanks for taking a look. The issue is:
```
>>> from setuptools.distutils import dir_util, log
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'setuptools.distutils'
```

Reply:
I see, it’s a not really a submodule, it’s a re-export.

But since `setuptools` just directly imports `distutils`, what’s the advantage of importing it from there instead of just doing `import distutils`?

If the intention is to to not import `distutils` because it’s deprecated, you need to actually not import it instead of importing it using a more roundabout path.

Reply:
These are good points, thanks for raising them.

> I see, it’s a not really a submodule, it’s a re-export.
> 
> But since `setuptools` just directly imports `distutils`, what’s the advantage of importing it from there instead of just doing `import distutils`?

I was looking to use the `setuptools` vendored `distutils` but think I have done this incorrectly. It's probably `setuptools._distutils`.

> If the intention is to to not import `distutils` because it’s deprecated, you need to actually remove the import instead of importing the same thing using a more roundabout path. Should be easy enough, there doesn’t seem to be any good reason to import either symbol from it: `log` is just a `logging.Logger` (you can just use another logger I guess) and `dir_util.mkpath(...)` provides the same functionality as `pathlib.Path(...).mkdir(parents=True, exists_ok=True)`

This patch is trying to do the absolute minimum necessary to make `pycc` not use `distutils` and instead use `setuptools`'s internal copy, and then to make sure that `setuptools` an optional runtime dependency. Given `pycc` is scheduled for deprecation in the next Numba release (xref: #8509), I'm not sure that it is worth spending time on finding a full replacement for `distutils`, then writing and testing it, though can probably make the changes RE: `log` and `mkpath` as noted.